lib/internal/process/promises.js

```diff
@@ -20,6 +23,14 @@ const pendingUnhandledRejections = [];
 const asyncHandledRejections = [];
 let lastPromiseId = 0;
 
+const states = {
+  none: 0,
+  warn: 1,
+  strict: 2
+};
+
+let state;
```

Comment:
There are 3 possible non-`undefined` states for `state` variable in this code.
4, including `undefined`.

But on the logic side, there are _5_ distinct possible states for `state` here:
1. Unknown (option not parsed yet)
2. Option is not set
3. Option is set to 'none'
4. Option is set to 'warn'
5. Option is set to 'strict'


Cases 1 and 2 should be distinct, but they both are represented as `state === undefined` here.

See also comment below on the same issue.

Reply:
I do not see `1` as a possibility: the state will be set as soon as we have a rejection.

Reply:
@BridgeAR No, that's a list potential states that `state` can represent in a given moment of time.

It's ok to represent "unknown" state as `state === undefined` (as the code now has), the problem was with it being mixed with "option is not set".